node/src/container_chain_spawner.rs

```diff
@@ -472,8 +481,9 @@ impl ContainerChainSpawner {
             if let Some(exit_notification_receiver) = maybe_exit_notification_receiver {
                 let _ = exit_notification_receiver.await;
             }
-
-            self.collation_cancellation_constructs = Some(f());
+        } else if self.collation_cancellation_constructs.is_none() {
```

Comment:
If you are going to use `collation_cancellation_constructs` to determine if collator process is not running, maybe we should rename it to indicate that.

Reply:
I can't think of a good name, but I added docs to the field definition.